### PR TITLE
s/EdgeKind::Const/EdgeKind::Static/

### LINKED ISSUE #150: Rename EdgeKind::Const -> EdgeKind::Static
Spec was updated in #108

### PR DESCRIPTION
**Pull request by aborgna-q**
Closes #150

### CODE REVIEW

File: src/hugr/validate.rs [hunk 1]
```diff
@@ -228,7 +228,9 @@ impl<'a> ValidationContext<'a> {
         // Input ports and output linear ports must always be connected
         let mut links = self.hugr.graph.port_links(port_index).peekable();
         let must_be_connected = match dir {
-            Direction::Incoming => port_kind.is_linear() || matches!(port_kind, EdgeKind::Const(_)),
+            Direction::Incoming => {
```
Comment (acl-cqc) [inline src/hugr/validate.rs:231]: Whoa, hang on. Drive-by I realize, but shouldn't this be `Direction::Incoming => true` ?

File: src/types.rs [hunk 2]
```diff
@@ -27,8 +27,8 @@ pub enum EdgeKind {
     ControlFlow,
     /// Data edges of a DDG region, also known as "wires".
     Value(SimpleType),
-    /// A reference to a constant value definition, used in the module region.
-    Const(ClassicType),
+    /// A reference to a static value definition, used in the module region.
```
Comment (acl-cqc) [inline src/types.rs:30]: Drive-by again, but I think now that we have local constants you could remove ", used in the module region" ?

File: src/types.rs [hunk 3]
```diff
@@ -46,16 +46,16 @@ impl EdgeKind {
...
 #[derive(Clone, Default, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
 pub struct Signature {
     /// Value inputs of the function.
     pub input: TypeRow,
     /// Value outputs of the function.
     pub output: TypeRow,
-    /// Possible constE input (for call / load-constant).
-    pub const_input: TypeRow,
+    /// Possible static input (for call / load-constant).
+    pub static_input: TypeRow,
```
Comment (acl-cqc) [inline src/types.rs:58]: Given we're renaming this, perhaps we should call it `static_inputs` as `TypeRow` allows more than one

Comment (acl-cqc) [reply src/types.rs:58]: Oh, no, sorry, we call inputs `input`, fair enough

File: src/hugr/validate.rs (hunk 1 above)
Comment (aborgna-q) [reply src/hugr/validate.rs:231]: Nop, incoming order ports can be disconnected.

**Review by acl-cqc — APPROVED** [review]
Looks good to me, thanks @aborgna-q . Just a couple of drive-bys, really...

File: src/hugr/validate.rs (hunk 1 above)
Comment (aborgna-q) [reply src/hugr/validate.rs:231]: Ah, but the check is still wrong as we are allowing more things besides order edges. I'll fix it.

Comment (acl-cqc) [reply src/hugr/validate.rs:231]: Oh, yes, we use ports for those in the implementation, but not the spec.

In which case let's update the comment two lines above; the word "ports" there must be interpreted as ports in the spec, not ports in the implementation.

The word we're looking for might be "dataflow" (at least the spec defines dataflow edges as value or static)
